app/user.go

```diff
@@ -981,6 +981,51 @@ func (a *App) UpdateActive(c request.CTX, user *model.User, active bool) (*model
 		}
 	}
 
+	dmChannels := []*model.Channel{}
+	channelsInTeam, aErr := a.GetChannelsForTeamForUser(c, "", user.Id, &model.ChannelSearchOpts{
+		IncludeDeleted: true,
```

Comment:
Don't we only need to query for archived channels in the case that we're unarchiving a user? If we do a conditional query based on whether we're restoring or deactivating a user we can avoid unnecessary record retrievals and remove some `DeleteAt == 0`/`DeleteAt != 0` conditionals below.

Reply:
Good catch!